### PR TITLE
Make trivial renames on registerwise classical operations possible

### LINKED ISSUE #1289: DecomposeBoxes not working with classicalexpbox
The following test fails with the latests release candidate.

```
def test_decompose_clexpbox() -> None:
    box_circ = Circuit(1)
    box_c = box_circ.add_c_register("c", 5)

    box_circ.add_classicalexpbox_register(box_c | box_c, box_c)  # type: ignore

    cbox = CircBox(box_circ)
    d = Circuit(1, 5)
    d.add_circbox(cbox, [0, 0, 1, 2, 3, 4])
    Transform.DecomposeBoxes().apply(d)
```

The cause of this is somewhere in https://github.com/CQCL/tket/commit/5e21d78a1c5254be8275b5c8519d1162d4271c5a

There is a PR for reverting this and adding a testcase for it at: https://github.com/CQCL/tket/pull/1287

### PR DESCRIPTION
**Pull request by cqc-alec**
# Description

Also add an early check to the `CircBox` constructor that the circuit is boxable.

# Related issues

Fixes #1289 .

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/tests/circuit_test.py [hunk 1]
```diff
@@ -1471,6 +1471,27 @@ def test_deserialization_from_junk() -> None:
...
 
+def test_decompose_clexpbox() -> None:
+    # https://github.com/CQCL/tket/issues/1289
+    c0 = Circuit()
+    c_reg = c0.add_c_register("c", 2)
+    c0.add_classicalexpbox_register(c_reg | c_reg, c_reg)  # type: ignore
+    cbox = CircBox(c0)
+    c = Circuit(0, 2)
+    c.add_circbox(cbox, [0, 1])
+    assert Transform.DecomposeBoxes().apply(c)
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/tests/circuit_test.py:1482]: I'll extend the test to make sure...

**Review by cqc-melf — APPROVED** [review]

**Comment by cqc-melf** [comment]
Could you make another release candidate that I could test again, just to make sure?

**Review by cqc-melf — APPROVED** [review]

**Comment by cqc-alec** [comment]
> Could you make another release candidate that I could test again, just to make sure?

Sure. Are you OK to ignore the coverage check failure? It is difficult to test this error case directly from C++.

**Comment by cqc-melf** [comment]
> > Could you make another release candidate that I could test again, just to make sure?
> 
> Sure. Are you OK to ignore the coverage check failure? It is difficult to test this error case directly from C++.

Yes that is fine with me

**Pull request merged**